File: src/core/validators/IncludeProperty/IncludePropertyMustNotContainIdentity.ts
import { ValidationRuleBase } from "../ValidationRuleBase";
export class IncludePropertyMustNotContainIdentity extends ValidationRuleBase<MetaEdGrammar.IncludePropertyContext>
{
    public isValid(context: MetaEdGrammar.IncludePropertyContext): boolean {
        return context.propertyComponents().propertyAnnotation().identity() == null;
    }
    public getFailureMessage(context: MetaEdGrammar.IncludePropertyContext): string {
        var topLevelEntity = context.GetAncestorContext<ITopLevelEntity>();
        return `Include property '${context.propertyName().GetText()}' is invalid to be used for the identity of ${topLevelEntity.EntityIdentifier()} '${topLevelEntity.EntityName()}'`;
    }
}

File: src/core/validators/IncludeProperty/IncludePropertyMustNotContainIdentity.cs
using MetaEd.Grammar.Antlr;
using MetaEd.Grammar.Antlr.Extensions;

namespace MetaEd.Core.Validator.IncludeProperty
{
    public class IncludePropertyMustNotContainIdentity : ValidationRuleBase<MetaEdGrammar.IncludePropertyContext>
    {
        public override bool IsValid(MetaEdGrammar.IncludePropertyContext context)
        {
            return context.propertyComponents().propertyAnnotation().identity() == null;
        }

        public override string GetFailureMessage(MetaEdGrammar.IncludePropertyContext context)
        {
            var topLevelEntity = context.GetAncestorContext<ITopLevelEntity>();
            return string.Format("Include property '{0}' is invalid to be used for the identity of {1} '{2}'", context.propertyName().GetText(), topLevelEntity.EntityIdentifier(), topLevelEntity.EntityName());
        }
    }
}